site/src/utils/filters.ts

```diff
@@ -1,3 +1,5 @@
-export const prepareQuery = (query?: string) => {
+export function prepareQuery(query: undefined): undefined;
+export function prepareQuery(query: string): string;
+export function prepareQuery(query?: string): string | undefined {
```

Comment:
```suggestion
export function prepareQuery<T extends string | undefined>(query: T): T extends undefined ? string | undefined : string {
```

as this is defined, I believe you'd actually be unable to pass a value of type `string | undefined`. could be "simplified"[^1] a bit to clear things up

[^1]: type parameters are not simple but neither is "overloading"

Reply:
So, I'm having trouble getting the compiler to be happy with your suggestion

I modified the overload signature to accept `string | undefined` values, but at this point, yeah, probably better to refactor this to have better type signatures down the line